controllers/courses.js

```diff
@@ -0,0 +1,136 @@
+const mongoose = require('mongoose')
+const { Users, Roles } = require('../models/users')
+const Courses = require('../models/courses')
+
+const courseController = {};
+
+/**
+ * @param {Object} req 
+ * @param {Object} res 
+ * @param {Function} next 
+ * @description Fetches all courses' data 
+ */
+courseController.getAll = async (req, res, next) => {
+    try {
+        const allCourses = await Courses.find()
+        const courses = allCourses.filter(course => {
+            return course.is_deleted === false
+        })
+        res.json(courses)
+    }
+    catch (err) {
+        res.send('Error' + err)
+    }
+};
+
+/**
+ * 
+ * @param {Object} req 
+ * @param {Object} res 
+ * @param {Function} next 
+ * @description Fetches one course
+ */
+courseController.getOne = async (req, res, next) => {
+    try {
+        let CourseId = mongoose.Types.ObjectId(req.params.id)
```

Comment:
don't need to check if the userId exists or not here simply use the findById query and if there is no user it will return null